src/mlpack/methods/kmeans/elkan_kmeans_impl.hpp

```diff
@@ -87,14 +90,13 @@ double ElkanKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
       // No change needed.  This point must still belong to that cluster.
       counts(assignments[i])++;
       newCentroids.col(assignments[i]) += arma::vec(dataset.col(i));
-      continue;
```

Comment:
We should keep this `continue`---otherwise the centroids will be computed twice for iterations where we take this `if`.